Updates and fixes.

rdebath opened the pull request:
These changes to fix layout and display issues
* e788436 Process CSI H with single argument
* f103061 Cursor postion and EOL wrapping.
* a47d9cd Add default tabs at 8 spaces.
* 1c63a86 Process CSI 90..107 m for bright colours
* 8ac2814 Turned on BCE

These avoid issues by hiding (sometimes invalid) known sequences.
* f4ca716 Discard obviously malformed UTF8
* 0170b60 Added code to ignore 256 and 16M colour sequences.
* 2ee6bb4 Eat OSC and other string sequences without interpretation.
* <s>093c250 If no current terminal, default to 80x25 screen</s>

Updates to improve features.
* 26b072a Minor fixes in character set conversion
* d2872bc Process CSI 8 m for trivial A_INVISIBLE
* cbf9414 Add DECALN
* 97f7608 Make user defined set "K" same as default.
* d8a8e80 Add ESC%G and ESC%@ to switch between UTF8 and ISO8859-1
* 34c371a Add linux palette sequences (OSC like).
* fe67c0d Delay opening GIF file until the first frame is parsed.
* 6b2bfda Produce commentary about GIF files without animation.
* 607a5e9 Compile the default font into the executable so it can be easily "installed".

rdebath committed:
Discard obviously malformed UTF8

rdebath committed:
Process CSI H with single argument

rdebath committed:
Process CSI 8 m for trivial A_INVISIBLE

Note: Linux console does not have CSI 8 m because it has no way to
highlight or "unconceal" the text without redrawing.

rdebath committed:
Cursor postion and EOL wrapping.

Various CSI cursor movements should be clipped and lots of other sequences
are supposed to clear the "wrap_needed" flag before they are processed.

rdebath committed:
Process CSI 90..107 m for bright colours

rdebath committed:
Added code to ignore 256 and 16M colour sequences.

Maybe these should be mapped to some default 16 colour map like the
current kernel does.

rdebath committed:
Turned on BCE

The code had BCE turned off when BCE was set true. Toggled.

rdebath committed:
Add default tabs at 8 spaces.

rdebath committed:
Minor fixes in character set conversion

rdebath committed:
Eat OSC and other string sequences without interpretation.

rdebath committed:
Add DECALN

rdebath committed:
Make user defined set "K" same as default.

rdebath committed:
Add ESC%G and ESC%@ to switch between UTF8 and ISO8859-1

rdebath committed:
Delay opening GIF file until the first frame is parsed.

This allows the start of the script file to alter rendering options
(specifically the global palette) and means that a single frame GIF
is simplified.

rdebath committed:
Add linux palette sequences.

rdebath committed:
Produce commentary about GIF files without animation.

rdebath committed:
Compile the default font into the executable

rather than a default font file name.

rdebath committed:
Don't try to write empty buffer at end of frame.

About 1 in 2000 frames fills the last block completely.

lecram commented:
Hi @rdebath!

Impressive work!

It'd be wonderful if you could open a different Pull Request for each set of related changes. For instance, "Compile the default font into the executable" should be a Pull Request on its own, since it's unrelated to the other commits. That way we can review, discuss and accept/reject each feature/fix separately. As far as I know, Pull Requests cannot be accepted partially.

About the improvements you've made to `term.c`: what kind of terminal are you trying to support? Currently we know that congif only supports the Linux console (terminfo `linux` terminal description). If we want to extend the set of sequences supported, it'd be nice to explicitly target some existing terminfo terminal description (say `vte` or `rxvt-unicode` or `dvtm`...). Alternatively, we can create a `congif` terminal description for terminfo, using `linux` as a base and extending it as we improve `term.c`.

rdebath committed:
Change defaults for terminal emulator size.

First Available from:
1) The command line.
2) From first line of script file
3) From the tty on file descriptor 0 (stdin).

The command line can be used to override just one of the values.

rdebath commented:
Okay, I've pulled some of the patches out into other pull requests, I'll rebase this one onto your master as & when so I can see what's still outstanding. It's pretty easy to `git cherry-pick` the patches you'd want to choose as long as there isn't a collision.

So far the emulation changes I've made are all targeting the Linux emulation (except the INVIS one which is just done trivially like ANSI.SYS). I suspect it's just changes that they've added to it since you last looked, not including the odd bug.

Most of my testing has been with files converted from old BBS stuff... 

![1992_acdu0892_SHARK ANS](https://user-images.githubusercontent.com/5853223/103107362-5cdb0280-4635-11eb-9a10-85b7c551403f.gif)
